mirascope/core/openai/stream.py

```diff
@@ -39,13 +44,54 @@ def cost(self) -> float | None:
 
     def _construct_message_param(
         self,
-        tool_calls: list[ChatCompletionMessageToolCallParam] | None = None,
+        tool_calls: list[ChatCompletionMessageToolCall] | None = None,
         content: str | None = None,
     ) -> ChatCompletionAssistantMessageParam:
-        message_param = ChatCompletionAssistantMessageParam(
+        """Constructs the message parameter for the assistant."""
+        message_param = ChatCompletionMessage(
             role="assistant",
             content=content,
         )
         if tool_calls:
-            message_param["tool_calls"] = tool_calls
-        return message_param
+            message_param.tool_calls = tool_calls
+        return message_param.model_dump(exclude={"function_call"})  # type: ignore
```

Comment:
remove type ignore and put type back to the assistant message for `message_param` I don't see why you changed it just to dump it?

Reply:
updated this to be more clear